src/shame.js

```diff
@@ -202,3 +202,11 @@ export const shamefullyDeleteRelatedAnalysisTransforms = (graphDiv, payload) =>
     }
   }
 };
+
+export const shamefullyAdjustSizeref = (gd, {update}) => {
+  const {'marker.size': size = null, 'marker.sizesrc': src = null} = update;
+  if (size && src) {
+    update['marker.sizeref'] = size.reduce((a, b) => Math.max(a, b)) / 2000;
+    update['marker.sizemode'] = 'area';
+  }
+};
```

Comment:
well, it seems like this is a good place to put that code in.
So we can never use size.mode diameter when there's a sizeref..?

Reply:
you can, you just have to set it yourself after. It's almost always the wrong thing to do tho, so I'm OK with forcing it back every time you switch that dropdown.